### PR TITLE
Make the ./b script keep rerunning itself so long as it thinks it can install a package

### PR DESCRIPTION
**Pull request by TrentHouliston**
Currently, it tells you that it installed and to try again. But we can automate that process and try again for you!

### CODE REVIEW

**Comment by TrentHouliston** [comment]
I now understand that this PR isn't fitting because #355 hasn't yet been merged

File: nuclear/b.py [hunk 1]
```diff
@@ -113,9 +114,8 @@
 
                             install_dependency(package)
 
-                            print()
-                            print("missing dependency installed, re-run command to continue")
-                            sys.exit(1)
+                            # Try rerunning ourself now the library exists
```
Comment (JosephusPaye) [inline nuclear/b.py:117]:
```suggestion
                            # Try re-running the current command now that the library exists
```

**Comment by KipHamiltons** [comment]
@Y0sh1dk pointed out to me that the python docs [recommend using subprocess.run() wherever possible](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module), instead of `.call()` and `.Popen()`. Is there a particular reason we use `.call()` ubiquitously?

**Comment by TrentHouliston** [comment]
Because most of this code was written before `subprocess.run` even existed. Additionally, the return type of call is the return type of the process that you execute. `run` on the other hand returns this wonderful fancy CompletedProcess instance which would require us to store it in a variable and then access the return value from it rather than just using the return value of the function.

That's the main reason it's being used over run in this case, so we can just exit with the return code

**Review by JosephusPaye — APPROVED** [review]
LGTM.

From Python 3.5, `subprocess.call()` [is equivalent](https://docs.python.org/3/library/subprocess.html#subprocess.call) to `subprocess.run(...).returncode`:

> Run the command described by _args_. Wait for command to complete, then return the `returncode` attribute.

The difference (`subprocess.call()` lacks the `check` and `input` parameters) is not relevant in this case.

**Pull request merged**